documentation/Get-Bootchart-metrics.md

```diff
@@ -0,0 +1,83 @@
+# Get Bootchart from a Clear Container#
+
+The following steps describe how to get a bootchart from a Clear Containers Container.
+Currently `systemd` is the default init process in the Clear Containers mini-os
+system. The Systemd project provides a tool called `systemd-bootchart` that
+collects the CPU load, disk load, memory usage, as well as per-process information
+from the os boot process.
+
+## Configure Clear Containers to get a bootchart##
+
+To get a bootchart from a Clear Container, follow the steps below.
+
+### Prepare your Clear Container image ###
+
+The file `data/cc-bootchart.conf` can be used get a bootchart graph.
+
+The file must be installed in a Clear Container image with `systemd` and `systemd-bootchart`.
+
+The file data/cc-bootchart can be installed with the configure option
+--with-cc-image-bootchart-config-dir=DIR DIR where DIR is any of the following paths.
+
+```
+/etc/systemd/bootchart.conf
+/etc/systemd/bootchart.conf.d/*.conf
+/run/systemd/bootchart.conf.d/*.conf
+/usr/lib/systemd/bootchart.conf.d/*.conf
+```
+
+Note: All these paths above refer to paths inside the container image.
+
+### Configure Clear Containers to boot with bootchart ###
+
+Make sure your kernel uses `systemd-bootchart` as the init process (PID 1).
+
+- Using a packaged version of `cc-oci-runtime`
+
+If you are using a packaged version of `cc-oci-runtime`, change
+`init=/usr/lib/systemd/systemd` to `init=PATH TO SYSTEMD-BOOTCHART BINARY`
+(usually  `init=/usr/lib/systemd/systemd-bootchart`) in the file
+`/etc/cc-oci-runtime/vm.json` (it can be copied from
+`/usr/share/defaults/cc-oci-runtime/vm.json` if it does not exist).
+
+```
+mkdir -p /etc/cc-oci-runtime/
+[ -f /etc/cc-oci-runtime/vm.json ] || \
+    cp /usr/share/defaults/cc-oci-runtime/vm.json /etc/cc-oci-runtime/vm.json
+sed -i 's,init=/usr/lib/systemd/systemd,init=/usr/lib/systemd/systemd-bootchart,g'\
+    /etc/cc-oci-runtime/vm.json
+```
+
+- Installing `cc-oci-runtime` from source code
+
+If `cc-oci-runtime` is installed from source code,
+change `init=/usr/lib/systemd/systemd` to `init=PATH TO SYSTEMD-BOOTCHART BINARY`
+in the file `data/kernel-cmdline` and then install.
+
+```
+sed -i 's,init=/usr/lib/systemd/systemd,init=/usr/lib/systemd/systemd-bootchart,g' \
+data/kernel-cmdline./autogen.sh
```

Comment:
Oops - looks like a missing newline as `./autogen.sh` should be on a separate line.